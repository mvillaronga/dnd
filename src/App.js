import React from 'react';
import './App.css';

import SpellSelectionPanel from './components/SpellSelectionPanel'


function App() {
  return (
    <div className="App">
      <header className="App-header">
        <SpellSelectionPanel />
      </header>
    </div>
  );
}

export default App;
