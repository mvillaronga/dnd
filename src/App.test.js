import React from 'react'
import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import App from './App'

configure({ adapter: new Adapter() });

const state = {
  selection: {
    spells: [
    ]
  },
  refs: {
    spells: {
      items: [
        {name: "magic missle", level:"1"},
      ]
    }
  },
  filter: {
    name: ''
  }
}

const mockStore = configureMockStore([thunk])
const store = mockStore(state)
const wrapper = () => mount(
  <Provider store={store}>
    <App />
  </Provider>
)

describe('App tests', () => {

  beforeEach(() => {
    store.clearActions();
  })

  it('should render the App', () => {
    const comp = wrapper()
    expect(comp.html()).toMatchSnapshot()
  })
})
