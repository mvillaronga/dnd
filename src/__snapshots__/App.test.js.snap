// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render the App 1`] = `"<div class=\\"App\\"><header class=\\"App-header\\"><div><h1>Spells</h1><hr><div>Filter&nbsp;<input type=\\"text\\"></div><hr><div><div></div></div><div><div>magic missle - 1</div></div></div></header></div>"`;
